src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -358,41 +358,67 @@ internal sealed class RSession : IRSession, IRCallbacks {
 
-                await LoadRtvsPackage(evaluation, libPath);
+                    await LoadRtvsPackage(evaluation, libPath);
 
-                if (!IsRemote && startupInfo.WorkingDirectory != null) {
-                    await evaluation.SetWorkingDirectoryAsync(startupInfo.WorkingDirectory);
-                } else {
-                    await evaluation.SetDefaultWorkingDirectoryAsync();
-                }
+                    if (!IsRemote && startupInfo.WorkingDirectory != null) {
+                        await evaluation.SetWorkingDirectoryAsync(startupInfo.WorkingDirectory);
+                    } else {
+                        await evaluation.SetDefaultWorkingDirectoryAsync();
+                    }
 
-                var callback = _callback;
-                if (callback != null) {
-                    await evaluation.SetVsGraphicsDeviceAsync();
-
-                    string mirrorUrl = callback.CranUrlFromName(startupInfo.CranMirrorName);
-                    await evaluation.SetVsCranSelectionAsync(mirrorUrl);
-
-                    await evaluation.SetCodePageAsync(startupInfo.CodePage);
-                    await evaluation.SetROptionsAsync();
-                    await evaluation.OverrideFunctionAsync("setwd", "base");
-                    await evaluation.SetFunctionRedirectionAsync();
-                    await evaluation.OptionsSetWidthAsync(startupInfo.TerminalWidth);
-
-                    try {
-                        // Only enable autosave for this session after querying the user about any existing file.
-                        // This way, if they happen to disconnect while still querying, we don't save the new empty
-                        // session and overwrite the old file.
-                        bool deleteExisting = await evaluation.QueryReloadAutosaveAsync();
-                        await evaluation.EnableAutosaveAsync(deleteExisting);
-                    } catch (REvaluationException) {
+                    var callback = _callback;
+                    if (callback != null) {
+                        await evaluation.SetVsGraphicsDeviceAsync();
+
+                        string mirrorUrl = callback.CranUrlFromName(startupInfo.CranMirrorName);
+
+                        try {
+                            await evaluation.SetVsCranSelectionAsync(mirrorUrl);
+                        } catch (REvaluationException ex) {
+                            await WriteErrorAsync(Resources.Error_SessionInitializationMirror, mirrorUrl, ex.Message);
+                        }
+
+                        try {
+                            await evaluation.SetCodePageAsync(startupInfo.CodePage);
+                        } catch (REvaluationException ex) {
+                            await WriteErrorAsync(Resources.Error_SessionInitializationCodePage, startupInfo.CodePage, ex.Message);
+                        }
+
+                        try {
+                            await evaluation.SetROptionsAsync();
```

Comment:
Some of those things aren't critical - say, if the codepage wasn't set properly, there might be bugs due to that (which is why it's still good to spell it out), but it doesn't really hinder any of our toolwindows from working properly. So I figured that it would be better to not block the user from using the product altogether for those errors - we should still get bug reports from the error messages that will be printed out. 

In contrast, failing to load the `rtvs` package is a fatal error. Some of these others, I wasn't sure about, and in all cases where I thought there was some chance of things breaking in a spectacular way down the line (i.e. some toolwindow not working or VS crashing), I didn't allow it to continue after those.